polymath/extensions/vector_ops.py

```
@@ -4,691 +4,707 @@
+            axis_for_show = axis
+            axis = (axis,)
+
+        # Check for duplicates
+        # Check for in-range values
+        selections = self._ndims * [False]
+        for i in axis:
+            try:
+                _ = selections[i]
+            except IndexError:
+                raise IndexError(f'axis is out of range ({-self._ndims},{self._ndims}) in '
+                                 f'{type(self)}.{op}: {i}')
```

Ruff (0.14.7)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)